apps/src/lib/ui/accounts/AddParentEmailController.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import color from '../../../util/color';
+import AddParentEmailModal from './AddParentEmailModal';
+
+/**
+ * This controller submits a hidden Rails-generated form that adds a parent
+ * email to an account.  It expects a jQuery wrapper for that form to be
+ * provided on construction. When the user submits a parent email,
+ * this controller loads the relevant information into the hidden Rails form
+ * and calls submit(). Rails injects all the JavaScript needed for the form to
+ * submit via AJAX with all the appropriate validation tokens, etc.
+ * The controller subscribes to events emitted by the Rails helper JavaScript
+ * to detect success or errors.
+ *
+ * Read more:
+ * http://guides.rubyonrails.org/working_with_javascript_in_rails.html#rails-ujs-event-handlers
+ * https://github.com/rails/jquery-ujs
+ */
+export default class AddParentEmailController {
+  /**
+   * @param {jQuery} form
+   * @param {jQuery} link
+   * @param {jQuery} displayedParentEmail
+   * @param {function(parentEmail:string, parentEmailOptIn:string)} parentEmailChangedCallback
+   */
+  constructor({form, link, displayedParentEmail}) {
+    this.form = form;
+    this.displayedParentEmail = displayedParentEmail;
+    link.click(this.showAddParentEmailModal);
+  }
+
+  showAddParentEmailModal = () => {
+    if (this.mountPoint) {
+      console.log('already have a mount point');
```

Comment:
Is this a debug statement you mean to remove before committing?

Reply:
I'll remove